src/ripple/basics/base_uint.h

```diff
@@ -50,6 +50,9 @@ class base_uint
     static_assert (Bits >= 64,
         "The length of a base_uint in bits must be at least 64.");
 
+    static_assert(sizeof(unsigned int) == 32 / CHAR_BIT,
```

Comment:
I agree that this is a good check, but wouldn't it be better to actually change the declaration of `pn` to `std::uint32_t` which expresses the intent and enforces the invariant?